### PR TITLE
Add Settings for license & automation analytics creds

### PR DESCRIPTION
**Pull request by rooftopcellist**
##### SUMMARY
Adds the necessary username and password needed for License & Automation Analytics.  

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API

### CODE REVIEW

File: awx/main/conf.py [hunk 1]
```diff
@@ -124,6 +124,44 @@ def _load_default_license_from_file():
...
 )
 
+register(
+    'CLOUD_REDHAT_USERNAME',
+    field_class=fields.CharField,
+    default='',
+    allow_blank=True,
+    encrypted=False,
+    read_only=True,
+    label=_('The RSA private key for SSH traffic to isolated instances'),
```
Comment (ryanpetrello) [inline awx/main/conf.py]: These labels and help text are wrong, let's update them or remove them.

File: awx/main/conf.py [hunk 2]
```diff
@@ -124,6 +124,44 @@ def _load_default_license_from_file():
     category_slug='system',
 )
 
+register(
+    'CLOUD_REDHAT_USERNAME',
+    field_class=fields.CharField,
+    default='',
+    allow_blank=True,
+    encrypted=False,
+    read_only=True,
```
Comment (ryanpetrello) [inline awx/main/conf.py]: I don't think `read_only` is what you want for any of these.

**Comment by rooftopcellist** [comment]
I just did some checks and each of these values can be changed, and the password is encrypted in the db.  

```
In [4]: settings.CLOUD_REDHAT_USERNAME = 'chadams'

In [5]: settings.AUTOMATION_ANALYTICS_URL = 'https://redhat.com'

In [6]: settings.CLOUD_REDHAT_PASSWORD = 'hello'
   ...: 

In [7]: username = getattr(settings, 'CLOUD_REDHAT_USERNAME', None)
   ...: password = getattr(settings, 'CLOUD_REDHAT_PASSWORD', None)
   ...: insights_url = getattr(settings, 'AUTOMATION_ANALYTICS_URL', None)
   ...: print(username, password, insights_url)
   ...: 
chadams hello https://redhat.com

```

```
  3 | 2019-08-28 14:18:01.995513+00 | 2019-08-28 14:18:01.995549+00 | CLOUD_REDHAT_PASSWORD       | "$encrypted$UTF8$AESCBC$Z0FBQUFBQmRabzBhNnpnZE9mN1RsaVRhdElvamdVbWNlV0p6d19tYmRKaUpybkpHZnZvT0dhM1RqajlFSlV1SVB
JWXMzYkJpSk1KU0hHTmtZQVEtbXY3d0hVU0pYaEZnbGc9PQ==" |   
```

**Comment by AlanCoding** [comment]
Looks like this would go in the UI at

https://github.com/ansible/awx/blob/e14f19468b4d55db0d121fc0fb7bfd090090629b/awx/ui/client/src/configuration/forms/system-form/sub-forms/system-misc.form.js

We might need to UI people to re-examine the organization of the setting categories with all the new ones we have been adding (note that UI categorizations are different from the API categories).

**Comment by ryanpetrello** [comment]
Since I need this for https://github.com/ansible/awx/pull/4611, I'd be fine getting this in *before* UI work happens.

File: awx/main/conf.py [hunk 3]
```diff
@@ -124,6 +124,44 @@ def _load_default_license_from_file():
...
 
+register(
+    'REDHAT_USERNAME',
+    field_class=fields.CharField,
+    default='',
+    allow_blank=True,
+    encrypted=False,
+    read_only=False,
+    label=_('Your Red hat customer username'),
+    help_text=_('Enter your Red Hat customer password to get a Tower License'),  # noqa
```
Comment (wenottingham) [inline awx/main/conf.py]: I would do "This username is used to retrieve license information and to send Automation Analytics".

File: awx/main/conf.py [hunk 4]
```diff
@@ -124,6 +124,44 @@ def _load_default_license_from_file():
...
+
+register(
+    'REDHAT_PASSWORD',
+    field_class=fields.CharField,
+    default='',
+    allow_blank=True,
+    encrypted=True,
+    read_only=False,
+    label=_('Your Red Hat customer password'),
+    help_text=_('Enter your Red Hat customer password to get a Tower License'),  # noqa
```
Comment (wenottingham) [inline awx/main/conf.py]: see above

File: awx/main/conf.py [hunk 5]
```diff
@@ -124,6 +124,44 @@ def _load_default_license_from_file():
...
 )
 
+register(
+    'REDHAT_USERNAME',
+    field_class=fields.CharField,
+    default='',
+    allow_blank=True,
+    encrypted=False,
+    read_only=False,
+    label=_('Red hat customer username'),
```
Comment (ryanpetrello) [inline awx/main/conf.py]: another minor nit, "Red Hat" is probably more correct

**Review by ryanpetrello — APPROVED** [review]

**Pull request merged**